webhook_server/libs/handlers/runner_handler.py

```
@@ -735,6 +735,79 @@ async def is_branch_exists(self, branch: str) -> bool:
+        the original PR owner's identity for DCO compliance.
+
+        Only amends if the original PR commits had a Signed-off-by.
+
+        Returns True if the commit was amended, False if no amendment was needed or possible.
+        """
+        # Get the original PR's commits to find the Signed-off-by
+        commits = await github_api_call(
+            lambda: list(pull_request.get_commits()), logger=self.logger, log_prefix=self.log_prefix
+        )
+        if not commits:
+            self.logger.debug(f"{self.log_prefix} No commits found in original PR, skipping author restore")
```

Ruff (0.15.10)

[warning] Logging statement uses f-string

(G004)